app/models/piece.rb

```diff
@@ -40,4 +40,24 @@ def set_image
     color ? color_string = "white" : color_string = "black"
     self.image_name ||= "#{color_string}-#{piece_type.downcase}.png"
   end
+
+  def move_to!(x, y)
+    @target = game.pieces.where(:x_position => x, :y_position => y).take
+    if @target.nil?
```

Comment:
would put an if statement after definition of `target` and before `if @target.nil?` to check for `valid_move?`